// path-to-regexp/dist/index.js

/**
 * Simple fixed path-to-regexp implementation
 * This is a minimal implementation that avoids the error
 */

// Export a simple function that just returns a regex
function pathToRegexp(path, keys, options) {
  // Handle string paths
  if (typeof path === 'string') {
    // Very simple implementation that just converts * to .* and escapes special chars
    path = path
      .replace(/[-\/\^$*+?.()|[]{}]/g, '\\$&')
      .replace(/\\\*/g, '.*');
    
    // Return a simple regex
    return new RegExp('^' + path + '$');
  }
  
  // If already a regex, just return it
  if (path instanceof RegExp) {
    return path;
  }
  
  // Default fallback
  return new RegExp('.*');
}

// Helper functions
pathToRegexp.parse = function(str) {
  return [{ name: null, prefix: '', suffix: '', pattern: '.*', modifier: '' }];
};

pathToRegexp.compile = function(str) {
  return function(params) { return str; };
};

pathToRegexp.tokensToRegExp = function() {
  return new RegExp('.*');
};

pathToRegexp.tokensToFunction = function() {
  return function() { return ''; };
};

module.exports = pathToRegexp;
module.exports.parse = pathToRegexp.parse;
module.exports.compile = pathToRegexp.compile;
module.exports.tokensToRegExp = pathToRegexp.tokensToRegExp;
module.exports.tokensToFunction = pathToRegexp.tokensToFunction;
